.circleci/generate.go

```diff
@@ -0,0 +1,146 @@
+
+version: 2.1
+
+orbs:
+  go: palantir/go@0.0.14
+  godel: palantir/godel@0.0.14
+
+jobs:
+  verify-circleci:
+    working_directory: /go/src/github.com/palantir/pkg
+    docker:
+      - image: "golang:1.13.4"
+    steps:
+      - checkout
+      - run: go version
+      - run: go run .circleci/generate.go .
+      - run: diff  <(cat .circleci/config.yml) <(go run .circleci/generate.go .)
+
+workflows:
+  version: 2
+  verify-test:
+    jobs:
+      - verify-circleci
+
+`
+
+	moduleTemplateContent = `      # {{.Module}}
+      - godel/verify:
+          name: {{.Module}}-verify
+          <<: *checkout-path
+          include-tests: true
+          executor:
+            name: go/golang
+            version: {{.CurrGoVersion}}
+            owner-repo: palantir/pkg/{{.Module}}
+      - godel/test:
+          name: {{.Module}}-test-go-{{.PrevGoMajorVersion}}
+          <<: *checkout-path
+          executor:
+            name: go/golang
+            version: {{.PrevGoVersion}}
+            owner-repo: palantir/pkg/{{.Module}}
+          requires:
+            - {{.Module}}-verify
+`
+)
+
+type TemplateObject struct {
+	Module             string
+	CurrGoVersion      string
+	PrevGoVersion      string
+	PrevGoMajorVersion string
+}
+
+func main() {
+	if len(os.Args) < 2 {
+		panic("parent directory must be provided as argument")
+	}
+	modParentDir := os.Args[1]
+	mods, err := modules(modParentDir)
+	if err != nil {
+		panic(err)
+	}
+	configYML, err := createConfigYML(mods, "1.13.4", "1.12.13")
+	if err != nil {
+		panic(err)
+	}
+	fmt.Print(configYML)
+}
+
+var moduleTemplate *template.Template
+
+func init() {
+	var err error
+	moduleTemplate, err = template.New("moduleTemplate").Parse(moduleTemplateContent)
+	if err != nil {
+		panic(fmt.Sprintf("failed to create moduleTemplate template: %v", err))
+	}
+}
+
+func createConfigYML(modDirs []string, currGoVersion, prevGoVersion string) (string, error) {
+	prevParts := strings.Split(prevGoVersion, ".")
+	if len(prevParts) < 2 {
+		return "", fmt.Errorf("prevGoVersion must have at least 2 parts separated by a period: %s", prevGoVersion)
+	}
+
+	outBuf := &bytes.Buffer{}
+	_, _ = fmt.Fprint(outBuf, header)
+	for i, modDir := range modDirs {
+		modJobs, err := moduleJobs(TemplateObject{
+			Module:             modDir,
+			CurrGoVersion:      currGoVersion,
+			PrevGoVersion:      prevGoVersion,
+			PrevGoMajorVersion: strings.Join(prevParts[:2], "."),
+		})
+		if err != nil {
+			return "", fmt.Errorf("failed to generate jobs for moduleTemplate %s: %v", modDir, err)
+		}
+		fmt.Fprint(outBuf, modJobs)
+		if i != len(modDirs)-1 {
+			fmt.Fprintln(outBuf)
+		}
+	}
+	return outBuf.String(), nil
+}
+
+func modules(parentDir string) ([]string, error) {
+	fis, err := ioutil.ReadDir(parentDir)
+	if err != nil {
+		return nil, fmt.Errorf("failed to read directory: %v", err)
+	}
+	var dirNames []string
+	for _, fi := range fis {
+		if !fi.IsDir() || strings.HasPrefix(fi.Name(), ".") {
```

Comment:
nit, probably doesn't matter in practice, but maybe we should verify there are .go files or go.mod so we don't include any directory that isn't a go module

Reply:
Yeah, when everything is a module we can check for `go.mod`, but right now a few of the projects are still non-module. Checking for `.go` would work for this repo, although in the general case the base package itself doesn't have to have a `.go` file (it could just have other packages within it). Will keep this for now given the one-off nature.